Tests/Basic/TemporaryFileTests.swift

```diff
@@ -0,0 +1,49 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+import Utility
+import Basic
+
+class TemporaryFileTests: XCTestCase {
+    func testBasicReadWrite() throws {
+        let filePath: String
+        do {
+            let file = try TemporaryFile(prefix: "myprefix", suffix: "mysuffix")
+            // Make sure the filename contains our prefix and suffix.
+            XCTAssertTrue(file.path.basename.hasPrefix("myprefix"))
+            XCTAssertTrue(file.path.basename.hasSuffix("mysuffix"))
+
+            // Check if file is created.
+            XCTAssertTrue(file.path.isFile)
+
+            // Try writing some data to the file.
+            let stream = OutputByteStream()
+            stream <<< "foo"
+            stream <<< "bar"
+            stream <<< "baz"
+            try fputs(stream.bytes.contents, file.fileHandle)
+
+            // Go to the begining of the file.
+            file.fileHandle.seek(toFileOffset: 0)
+            // Read the contents.
+            let contents = String(data: file.fileHandle.readDataToEndOfFile(), encoding: NSUTF8StringEncoding)
+            XCTAssertEqual(contents, "foobarbaz")
+
+            filePath = file.path
+        }
+        // File should be deleted now.
+        XCTAssertFalse(filePath.isFile)
+    }
+
+    static var allTests = [
+        ("testBasicReadWrite", testBasicReadWrite),
+    ]
+}
```

Comment:
It would be good to have a test that verifies that the temporary files come up with a true temporary path, maybe as simple as just verifying two calls get different paths when both haven't been deleted.